core/src/test/java/org/infinispan/protostream/impl/TagWriterImplTest.java

```diff
@@ -22,7 +23,57 @@ public class TagWriterImplTest {
+
+   @Test
+   public void testUTF8Compatibility() throws IOException {
+      SerializationContext ctx = ProtobufUtil.newSerializationContext();
+      testUTF8String(ctx, utf8Str(10, false));
+      testUTF8String(ctx, utf8Str(100, false));
+      testUTF8String(ctx, utf8Str(1000, false));
```

Comment:
I would add an extra test with random length
```java
testUTF8String(ctx, utf8Str(ThreadLocalRandom.current().nextInt(100000000), false));
```
ps log the value just in case it fails we know the length to use :smile: